File: C++/SelectionSort/src/Book.cpp
#include "Book.h"

Book::Book(string _title, Author _author, int _numberPages) : author(_author) {
	title = _title;
	numberPages = _numberPages;
}

int Book::compareAuthor(Book _book) {
	return this->author.compareTo(_book.getAuthor());
}

int Book::compareNumberPages(Book _book) {
	int compare = 0;

	if(this->numberPages < _book.getNumberPages()) {
		compare = -1;
	} else if(this->numberPages > _book.getNumberPages()) {
		compare = 1;
	}

	return compare;
}

int Book::compareTitle(Book _book) {
	int compare = 0;

	if(this->title < _book.getTitle()) {
		compare = -1;
	} else if(this->title > _book.getTitle()) {
		compare = 1;
	}

	return compare;
}

File: Python/SelectionSort/src/Book.py

from Author import Author

class Book:
    def __init__(self, title = "", author = Author("",""), numberPages = 0):
        self.__title = title
        self.__author = author
        self.__numberPages = numberPages

    def compareAuthor(self, book):
        return self.__author.compareTo(book.getAuthor())

    def compareNumberPages(self, book):
        compare = 0
        if int(self.__numberPages) < int(book.getNumberPages()):
            compare = -1
        elif int(self.__numberPages) > int(book.getNumberPages()):
            compare = 1
        return compare

    def compareTitle(self, book):
        compare = 0
        if self.__title < book.getTitle():
            compare = -1
        elif self.__title > book.getTitle():
            compare = 1
        return compare

    def getAuthor(self):
        return self.__author

    def getNumberPages(self):
        return self.__numberPages

    def getTitle(self):
        return self.__title
